pkg/api/v1alpha1/cluster_defaults.go

```diff
@@ -50,6 +51,14 @@ func setWorkerNodeGroupDefaults(cluster *Cluster) error {
 		logger.V(1).Info("First worker node group name not specified. Defaulting name to md-0.")
 		cluster.Spec.WorkerNodeGroupConfigurations[0].Name = "md-0"
 	}
+
+	for i := range cluster.Spec.WorkerNodeGroupConfigurations {
+		w := &cluster.Spec.WorkerNodeGroupConfigurations[i]
+		if w.Count == nil {
```

Comment:
yeah, [this follow up PR will fix it.](https://github.com/aws/eks-anywhere/pull/3583/files#diff-6fecc4a57293bf395bf01de6b88ac3e4a291551d8c8a651421097cdbc021353aR55) I'm trying to keep the two changes as separate as I can because this one is so big. Maybe not worth it, but helps keep my thinking clearer.